code/modules/clothing/spacesuits/rig/modules/ds13/stasis.dm

```diff
@@ -0,0 +1,67 @@
+/obj/item/rig_module/mounted/stasis
+	name = "Stasis Module"
+	interface_name = "Stasis Module"
+	desc = "Stasis module is capable of producing a temporary time dilation, making objects move at an extremely slow rate for a period of time."
+	interface_desc = "Stasis module is capable of producing a temporary time dilation, making objects move at an extremely slow rate for a period of time."
+	icon = 'icons/obj/rig_modules.dmi'
+	icon_state = "module"
+
+	base_type = /obj/item/rig_module/mounted/stasis
+
+	selectable = 1
+	usable = 0
+	process_with_rig = FALSE
+
+	module_cooldown = 0
+
+	gun = /obj/item/weapon/gun/energy/stasis
+	suit_overlay = "stasismodule"
+	second_overlay = "stasisbar_100"
+	second_overlay_layer = EYE_GLOW_LAYER
+	second_overlay_plane = EFFECTS_ABOVE_LIGHTING_PLANE
+	second_overlay_flags = KEEP_APART
+
+/datum/proc/update_stas_charge()
+	return
+
+/obj/item/rig_module/mounted/stasis/installed()
+	..()
+	holder.stasis = src
+
+/obj/item/rig_module/mounted/stasis/uninstalled()
+	..()
+	holder.stasis = null
+
+/obj/item/rig_module/mounted/stasis/update_stas_charge()
+	var/percentage
```

Comment:
put calls to this proc inside others, instead of manually calling it with others at the same time